### PR TITLE
fix: Remove flicker when selecting different engines for displaying their URLs.

### PR DESCRIPTION
**Pull request by Standard8**
This avoids clearing the table on every reload and instead updates the existing table if there is one.

### CODE REVIEW

File: extension/content/engineUrlView.mjs [hunk 1]
```diff
@@ -19,22 +25,23 @@ export default class EngineUrlView extends HTMLElement {
...
+  createTableFragment(colHeaders, rowHeaders) {
     let fragment = document.createDocumentFragment();
-    let table = document.createElement("table");
-    let thead = table.createTHead();
-    let tbody = table.createTBody();
+    this.#table = document.createElement("table");
+    let colgroup = document.createElement("colgroup");
+    colgroup.appendChild(document.createElement("col"));
+    colgroup.appendChild(document.createElement("col"));
+    this.#table.appendChild(colgroup);
```
Comment (mandysGit) [inline extension/content/engineUrlView.mjs:41]: The colgroup is not doing anything functional in the code. From what I can tell, perhaps they were created to have a label for each column. But it seems unnecessary to have these unless they're being used somewhere else. 

I think it's safe to remove the colgroup and associated <col> elements from the code to simplify it without impacting the table's functionality or appearance.

File: extension/content/engineUrlView.mjs [hunk 2]
```diff
@@ -43,21 +50,30 @@ export default class EngineUrlView extends HTMLElement {
...
       let cell = row.insertCell(-1);
-      let url = sortedUrls[index];
+      cell.textContent = "";
+    });
+
+    fragment.appendChild(this.#table);
+    this.appendChild(fragment);
+  }
+
+  updateUrls(header, sortedUrls) {
```
Comment (mandysGit) [inline extension/content/engineUrlView.mjs:65]: For `updateUrls` function, could we split it up into two? Reason: updating the header does not update a url, it only updates the engine name. 

Maybe something like this to make it more clear: 
```js
#updateHeader(header) {
  this.#table.tHead.rows[0].cells[1].textContent = header;
}

#updateUrls(sortedUrls) {
  let tBody = this.#table.tBodies[0];

  sortedUrls.forEach((url, index) => {
    let cell = tBody.rows[index].cells[1];
    if (url) {
      cell.replaceChildren(this.createAnchor(url));
    } else {
      cell.textContent = "Not Specified";
    }
  });
}
```

**Review by mandysGit — CHANGES_REQUESTED** [review]

**Comment by mandysGit** [comment]
Thank you. The overall implementation looks good and I tested the flickering is gone now. Just requesting those two code changes above.

File: extension/content/engineUrlView.mjs (hunk 1 above)
Comment (Standard8) [reply extension/content/engineUrlView.mjs:41]: I think I might have originally had it for css purporses, but looks like it isn't required now.

File: extension/content/engineUrlView.mjs (hunk 2 above)
Comment (Standard8) [reply extension/content/engineUrlView.mjs]: I'm not sure there's a need to split this into two, I do think I can improve the function naming though.

**Review by mandysGit — APPROVED** [review]
Thank you.

**Pull request merged**